### DISCUSSION TITLE
Dump of stuff for lazer.

### DISCUSSION THREAD

**Discussion by freqawi** [ppy/osu#28912] [Ideas]
Hello!

I tried to do this awhile back but a power outage made me lose progress on all that I had written.

I know some stuff likely already had been shared but they are so important to me that I'd like to add them here.

### **Fixes/Changes.** 

1. **Song Select** is sorted from Top->Bottom compared to Bottom->Top for stable. When adding new songs or sorting songs you already have it feels backwards compared to stable, (because it is), It isn't the biggest gripe as i've learned it but players moving from stable will be confused just as I was I'm sure.

2. ~~**The chat.** Oh god the chat. I don't think that it's at all an understatement in it's current state that its plain unusable. I have no clue what the sorting is for direct messages is, it doesn't follow A-Z, Z-A or even what I would want, Oldest to Recent message. I won't even use it and will just click the message notification to jump to a message. please fix.~~  **Fixed 9/12/2024** Thank you so much. <3

3. This might just be me but I couldn't find 'Currently Online' for a long time of using lazer, until last time I tried to write this where apparently I opened my eyes and it was where the friends was. However, you cannot **search for offline players** like you would the on a browser. It would be great if it could, or maybe a have a new tab introduced for global search of players.

4. **Saving maps in a mapset** that have scores locally from being deleted when updating a mapset. I have wanted this on stable for the longest time, since maps can update with only a few changes but it will still delete the score I want to keep. I often will not update mapsets if there is a score I like on it because of this.

5. **Collections** are quite a pain on lazer. On stable, if you wanted to add a song to a collection you would simply make it in the same interface that you add songs to your collections, simple and easy. On lazer, you have to make the collection in a different interface then add the song to it on a seperate interface. Not a very big deal but IMO stable does this way better.

6. **Pinning.** There are 2 things at the moment I can think of wanting pinned in place. One of which being the "Sort By" selections, I like to always be sorted by Title, but if I don't, at the end of a session, select to be sorted by title again, it will disappear into the drop down. i think it would be great to be able to pin a select few of the sort by options to the start for quick easy access. Similarly for the skin selection, pinning skins you'd frequently change or swap to at the top so we don't have to keep searching in, for some people, endless sea of skins. I know likely a large overhaul of the skin selection is going to happen but still. pinnables.

7. **Anti Chatter and Smoothing** settings to Osu's tablet settings. Probably don't need to elaborate on this, just add more functionality to the tablet settings like one would do in OpenTabletDriver.

8. **Show Score Grading from submitted global scores** on the song select, Stable does this why doesn't lazer, often I am playing a map thinking I don't already have an Full Combo on it only to check Global and see that actually do.

9. ~~**Importing a specific type of thing from a stable install**. I've done osu stable import a few times before and some scores still do not transfer over, but every time I do it, it does everything, maps, skins, scores. adding functionality to do one of them only would be great~~, or even importing from differences between stable and lazer players, say if you play on both clients and download new maps to stable, I should be able to only transfer the new additions.

I was going to add some stuff on skinning but that's most definitely going to be overhauled so not here at least.

### **New Stuff.**

1. **Ability to spectate stable from lazer**, I dont even know if this is physically possible but I think one day it would be great, also seeing stable players in online menu and their currently playing song. like on stable.

2. **Cloud saves** for user settings. I have gone through many installs I think it would be great if you can set your current settings as default when you sign in, or even for LAN players like at Roundtable just sign in and everything on osu is the way you like it.

### **Mod Ideas.**

**Not So Auto.** in my imagination is a mod that would display a second cursor and would automatically aim to every second note, but not tap. So basically you would have to not only aim the correct notes but also be watching the other cursor and tap for both. Very challenging, (At least in theory*), but I think it would be very fun.

I might of forgotten some of the ideas or fixes I had, I'll make a new thread if I think of them (and if they haven't already been suggested.
I also don't mean to offend anyone with these, I am sure all osu dev's do a lot of work to make lazer what it is currently and what it will be in the future, they are just things that I believe would make lazer the perfect stable replacement. (In my eyes.)

<3

**Comment by cihe13375** [comment]
> 9. Importing a specific type of thing from a stable install

I think this is already possible?
![aaaa](https://github.com/user-attachments/assets/410a3318-c1f0-4eec-896a-97aa8f15ef25)

**Reply by freqawi** [reply]
is this new what?

**Reply by cihe13375** [reply]
idk but I think it has been possible since long before ...?

**Reply by freqawi** [reply]
maybe i am just very stupid, blind, or both. but also having it know that there were changes from osustable directory and asking to import the new files is still a good idea.

**Comment by peppy** [comment]
>  I don't think that it's at all an understatement in it's current state that its plain unusable. I have no clue what the sorting is for direct messages is, it doesn't follow A-Z, Z-A or even what I would want, Oldest to Recent message. I won't even use it and will just click the message notification to jump to a message. please fix.

Should be most-recent-at-top. Is for me, at least.

**Reply by freqawi** [reply]
I haven't messed with any settings regarding chat, so I am unsure of what the issue is and assumed it was a lazer thing.

**Reply by marcostudios** [reply]
It might be me archiving DMs by not closing them, but it orders it only on the first message and does not update on the others. Also in case of a big list it might be nice to have a way to directly search from the list instead of having to find the profile and clicking on the message button.

**Reply by freqawi** [reply]
My dms have only about 15 users.. unless I misunderstood you.

**Reply by marcostudios** [reply]
Sorry i was saying my experience, i should've wrote a new discussion instead of replying on one tbh
